util-ref/README.md

```diff
@@ -0,0 +1,66 @@
+# Util-Ref
+
+This project provides classes and utility wrappers for interactions with data storage and are used throughout [CAF](http://cafapi.github.io/) component and service development.
+
+## DataSource
+
+Interface for defining how to retrieve objects or streams from a reference.
+
+## DataSourceException
+
+General exception for DataSource acquisition failures.
+
+## ReferencedData
+
+Utility wrapper for allowing data to potentially be within a message or located on a remote DataStore. The acquire(ObjectSource) method allows transparent method of obtaining an InputStream to the data.
+
+This is primarily intended for use with large amounts of binary data that needs to be processed without storing it all in memory.
+
+### Usage
+
+To instantiate a ReferencedData object, call one of the methods:
+
+- `public static ReferencedData getReferencedData(final String ref)`
+
+- `public static ReferencedData getWrappedData(final byte[] data)`
+
+- `public static ReferencedData getWrappedData(final String ref, final byte[] data)`
+
+The `acquire(...)` method is used to return the referenced data as a stream, and can potentially perform a remote lookup if the object has only member value `reference`.
+
+## ReferencedObject
+
+Utility wrapper for allowing data to potentially be within a message or located on a remote DataStore. The acquire(ObjectSource) method allows transparent method of obtaining the wrapped data, which will only be retrieved the first time acquire is called (if it is not already present).
```

Comment:
"allowing data" is the same way that `ReferencedData` was described - should this not be "allowing a serialised object" or something like that?